provider/core/src/response.rs

```diff
@@ -108,6 +115,19 @@ impl Cart {
     }
 }
 
+/// An empty cart sentinel value for the DataPayload. Better for memory
+/// layout than Option<Cart> in the DataPayload; see #4438.
+///
+/// Get one with the [`empty_cart()`] function.
+static EMPTY_CART: OnceCell<Cart> = OnceCell::new();
```

Comment:
nit: this can be a local variable in `empty_cart`, then you don't have two items

Reply:
Fixed via moving the sentinel into `refcount.rs` and making it a `const`